databuilder/publisher/neo4j_csv_publisher.py

```diff
@@ -318,10 +320,26 @@ def _publish_relation(self, relation_file: str, tx: Transaction) -> Transaction:
 
         with open(relation_file, 'r', encoding='utf8') as relation_csv:
             for rel_record in pandas.read_csv(relation_csv, na_filter=False).to_dict(orient="records"):
-                stmt = self.create_relationship_merge_statement(rel_record=rel_record)
-                params = self._create_props_param(rel_record)
-                tx = self._execute_statement(stmt, tx, params,
-                                             expect_result=self._confirm_rel_created)
+                badge_exception = True
+                retries_for_badge_exception = 10  # TODO not sure how many times to retry
+                while badge_exception and retries_for_badge_exception > 0:
+                    try:
+                        stmt = self.create_relationship_merge_statement(rel_record=rel_record)
+                        params = self._create_props_param(rel_record)
+                        tx = self._execute_statement(stmt, tx, params,
+                                                     expect_result=self._confirm_rel_created)
+                        badge_exception = False  # if no exception happens we stop inner loop
+                        # TODO use break instead ^ ?
+                    except TransientError as e:
+                        # if exception is due to badge relation issue TODO could use a specific
+                        # exception for this condition (using TransientError)
+                        if rel_record[RELATION_START_LABEL] == BadgeMetadata.BADGE_NODE_LABEL\
```

Comment:
I need to think a bit more on this code. Also we should put the badgemetadata through config(a list of node label) instead of hard code here, so that in the future we could handle something similar for other type of metadata if needed.

Reply:
@feng-tao I think i fixed this, please let me know if this is what you meant.

Reply:
will take a look